packages/sdk/src/interaction/interaction.js

```diff
@@ -149,9 +159,19 @@ export const prepAccount = (acct, opts = {}) => ix => {
   const role = opts.role
   const tempId = uuid()
 
-  if (acct.authorization && isFn(acct.authorization)) acct = {resolve: acct.authorization}
+  if (acct.authorization && isFn(acct.authorization))
+    acct = {resolve: acct.authorization}
   if (!acct.authorization && isFn(acct)) acct = {resolve: acct}
 
+  const resolve = acct.resolve
+  if (resolve)
+    acct.resolve = async acct =>
+      await [resolve, prepAccountKeyId].reduce(
```

Comment:
```suggestion
      [resolve, prepAccountKeyId].reduce(
```

No need to await the return value of an async function.